### PR TITLE
feat: Add notHasProperty expectation

### PR DESCRIPTION
**Pull request by maikeru**
###What
Makes it possible to assert that a property does not exist in the body of the response.

###Why
This is a change I made locally when running artillery against a graphql endpoint. Graphql will as a rule respond with a 200 even if some part of the query fails. Errors are contained in an errors array within the json response.

Here's an example failure:
```
{
  "data": null,
  "errors": [
    {
      "message": "Field 'repository' is missing required arguments: name",
      "path": [
        "query",
        "repository"
      ]
    }
}
```

If the request succeeds the response will also be a 200 but `data:` will contain an object and `errors:` will not exist. It's not possible to detect this with the existing expectations so I added a new one.

### CODE REVIEW

**Comment by hassy** [comment]
Nice! Could we implement `notHasProperty` via a call to existing `hasProperty` function to simplify the code (rather than duplicating code).

**Comment by maikeru** [comment]
Makes sense, I'll give that a try.

**Comment by maikeru** [comment]
@hassy I ended up extracting the common code from `hasProperty` and `notHasProperty` as simply calling `hasProperty` meant that I had to transform both the arguments to `hasProperty` and the result object that came back in a way that made the code hard to follow. If you would like to see how it turned out you can see it [here](https://github.com/artilleryio/artillery-plugin-expect/compare/master...maikeru:wrap-has-property?expand=1).

Also added some tests for `hasProperty` and `notHasProperty` to make sure the refactor didn't break anything.

**Comment by maikeru** [comment]
@hassy Any further thoughts on this PR?

**Comment by 72lions** [comment]
@hassy what do you think about this PR?

**Comment by hassy** [comment]
This is a great PR, thanks @maikeru! Apologies for not merging sooner!
